src/realm/object-store/sectioned_results.cpp

```diff
@@ -401,6 +401,14 @@ void SectionedResults::calculate_sections()
             m_row_to_index_path[i] = {section.index, section.indices.size() - 1};
         }
     }
+    if (!has_performed_initial_evalutation) {
+        m_previous_key_to_index_lookup.clear();
+        m_prev_section_index_to_key.clear();
```

Comment:
I think it might make more sense to assert that these are empty rather than clearing them? If they're non-empty then something's gone wrong.